core/vibes/soul/lib/streamable.tsx

```diff
@@ -61,6 +65,12 @@ const promiseCache = weakRefCache<string, Promise<unknown>>();
 function all<T extends readonly unknown[] | []>(
   streamables: T,
 ): Streamable<{ -readonly [P in keyof T]: Awaited<T[P]> }> {
+  // Avoid creating an unnecessary promise if all values are already resolved
+  if (streamables.every(isResolved)) {
```

Comment:
If the `streamables` argument is stable, `Streamable.all` will correctly cache the (extra) promise below. However, when `streamables` doesn't have a stable identity, but contains only resolved promises, wrapping it with `Streamable.all` exacerbates incidental rerenders by adding a `Suspense`-induced flicker that wouldn't be otherwise present.

One can argue that this is a good thing, as it makes an existing performance issue much more visible, but, all things considering, I think we're better off fixing it.

Reply:
I'll follow up with a corresponding PR for this in VIBES.

Reply:
https://github.com/makeswift/vibes/pull/617